exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -86,6 +92,19 @@
 
   BooleanValidator EXTERNAL_SORT_DISABLE_MANAGED_OPTION = new BooleanValidator("exec.sort.disable_managed", false);
 
+  // Hash Aggregate Options
+
+  String HASHAGG_NUM_PARTITIONS_KEY = "drill.exec.hashagg.num_partitions";
```

Comment:
Changed all three to have both the config option ("drill.exec.hashagg....") and its matching _KEY (same, sans the prefix "drill.") which is only used to create the validator.
Hence all three can be set either in the config file, or for the session: 
Number of partitions, max memory, and minimum batches per partitions.